runner/src/diffs.rs

```diff
@@ -44,17 +44,25 @@ impl ModifiedContent {
     }
 }
 
-// Get all the files that have been modified in the commits specified by `commits`. Does not include
-// new files, unless they are added to git. If present, `commits.commits` must be a positive number.
-// If it is zero or negative, only the last commit will be considered for finding the modified
-// files. If `commits.commits` is not present, all files will be considered.
-pub fn modified_files(commits: &Commits) -> ModifiedContent {
-    let files = commits.commits.map(|commits| {
+/// Get all the files that have been modified in the given `scope`. If the scope is `Scope::All`
+/// returns an empty list of files in `ModifiedContent`. For `Scope::Commits` and
+/// `Scope::DiffToMain` the returned files include tracked files with unstaged changes in, but do
+/// not include new files that are not tracked (i.e., not added to git yet). For
+/// `Scope::Commits(N)`, a positive number of commits must be given. In case of zero or
+/// a negative number, only the last commit will be considered for finding the modified files.
+pub fn modified_files(scope: &Scope) -> ModifiedContent {
+    let args = match scope {
+        Scope::All => None,
+        Scope::Commits(commits) => Some(vec![format!("HEAD~{}", std::cmp::max(1, *commits))]),
```

Comment:
I think these can just be flattened to Vec, and return empty Vec for Scope::All. The extra optionality does not seem to represent any actual state.

Reply:
You are right.  But in the end `files` is `Option`. So, having `args` as option gives a nicer mapping.

I've removed `vec` for now. It will be needed if we manage to update the git version and use the additional `--merge-base` flag.